pennylane/transforms/metric_tensor.py

```diff
@@ -250,30 +227,43 @@ def circuit(weights):
 
         qnode = qnode.qnodes.qnodes[0]
 
-    def _metric_tensor_fn(*args, **kwargs):
-        jac = qml.math.stack(qml.transforms.classical_jacobian(qnode)(*args, **kwargs))
-        jac = qml.math.reshape(jac, [qnode.qtape.num_params, -1])
+    mt_fn = self.default_qnode_wrapper(qnode, targs, tkwargs)
+    cjac_fn = qml.transforms.classical_jacobian(qnode)
+
+    def wrapper(*args, **kwargs):
+        mt = mt_fn(*args, **kwargs)
 
-        wrt, perm = np.nonzero(qml.math.toarray(jac))
-        perm = np.argsort(np.argsort(perm))
+        if not hybrid:
+            return mt
+
+        cjac = cjac_fn(*args, **kwargs)
+
+        if isinstance(cjac, tuple):
+            if len(cjac) == 1:
+                cjac = cjac[0]
+            else:
+                # Classical processing of multiple arguments is present. Return mt @ cjac.
+                metric_tensors = []
 
-        qnode.construct(args, kwargs)
+                for c in cjac:
+                    if c is not None:

```

Comment:
how can `None` values appear in the classical jacobian?

Reply:
depending on the autodiff framework, `None` appears if the corresponding QNode argument is non-differentiable:

```python
@qml.qnode(dev)
def circuit(x, y):
    qml.RX(2*x, wires=0)
    qml.RY(y**3, wires=0)
    return qml.expval(qml.PauliZ(0)

>>> x = np.array(0.4, requires_grad=True)
>>> y = np.array(0.4, requires_grad=False)
```